heat/nn/data_parallel.py

```diff
@@ -0,0 +1,275 @@
+import bisect
+import functools
+import heat as ht
+import operator
+import torch
+import torch.nn as tnn
+
+from collections import OrderedDict
+from typing import Callable, List
+
+from heat.core.communication import MPI
+
+__all__ = ["DataParallel"]
+
+
+class DataParallel(tnn.Module):
+    """
+    Implements data parallelism across multiple processes. This means that the same model will be run locally
+    on each process. Creation of the model parallels to PyTorch, the only changes are using HeAT layers (ht.nn.layer)
+    in the initialization of the network. I.E. (example code contains)
+    .. code-block:: python
+
+        class TestingModel(torch.nn.Module):
+            def __init__(self):
+                super(TestingModel, self).__init__()
+                self.net1 = ht.nn.Linear(10, 10)
+                self.relu = ht.nn.ReLU()
+                self.net2 = ht.nn.Linear(10, 5)
+
+            def forward(self, x):
+                return self.net2(self.relu(self.net1(x)))
+
+        ht_model = ht.nn.DataParallel(TestingModel(), comm, optimizer)
+
+    and a requirement of giving a HeAT communicator (``comm``) as well as a Torch optimizer (``optimizer``). For the
+    given model both the ``__init__()`` and ``forward()`` functions must be defined in the class defining the network.
+
+    It is highly recommended that a HeAT DataLoader is used, see :func:`..utils.data.datatools.DataLoader`. The
+    default communications scheme for this is blocking. The blocking scheme will average the model parameters during
+    the backwards step, synchronizing them before the next model iteration.
+
+    To use the non-blocking communications for parameter updates, negate the optional flag ``blocking``.
+
+    Attributes
+    ----------
+    module : torch.nn.Module
+        the local module
+    comm : heat.MPICommunicator
```

Comment:
remove "heat.", add direct cross-ref (also, does it necessarily have to be the MPICommunicator, or a generalized ht communicator ? )

Reply:
the MPICommunicator is the general heat communicator